### PR TITLE
Adding support to enable remote management

### PR DESCRIPTION
**Pull request by mtandon80**
Adding support to enable remote management for already registered Azure Stack. Following changes done as part or enabling remote management
1. Call action "EnableRemoteManagement" on existing registration resource
2. Upsert linked subscription resource on customer Azure subscription
2. Enable-AzsCloudConnection PEP cmdlet will be called to actually enable remote management on AzureStack Hub

### CODE REVIEW

File: Registration/RemoteManagement.psm1 [hunk 1]
```diff
@@ -0,0 +1,489 @@
...
+
+    Log-Output "Resource creation params: $(ConvertTo-Json $resourceCreationParams)"
+
+    do
+    {
+        try
+        {  
+            $DebugPreference="Continue"                       
+            Log-Output "Creating resource group '$ResourceGroupName' in location $ResourceGroupLocation."
+            $resourceGroup = New-AzureRmResourceGroup -Name $ResourceGroupName -Location $ResourceGroupLocation -Force
```
Comment (bganapa) [inline Registration/RemoteManagement.psm1:299]: pls add -ErrorAction Stop

File: Registration/RemoteManagement.psm1 [hunk 2]
```diff
@@ -0,0 +1,489 @@
...
+    $currentAttempt = 0
+    $maxAttempt = 3
+    $sleepSeconds = 10
+    do
+    {
+        try
+        {
+            Log-Output "Registering Azure Stack resource provider."
+            [Version]$azurePSVersion = (Get-Module AzureRm.Resources).Version
+            if ($azurePSVersion -ge [Version]"4.3.2")
```
Comment (bganapa) [inline Registration/RemoteManagement.psm1:28]: you can do(gcm Register-AzureRmResourceProvider).Parameters.ContainsKey('Force')) instead of version check

**Comment by Justinha** [comment]
@myoungerman

File: Registration/RemoteManagement.psm1 (hunk 2 above)
Comment (mtandon80) [reply Registration/RemoteManagement.psm1]: this is same as registration code. Like to reuse existing implementation

File: Registration/RemoteManagement.psm1 [hunk 3]
```diff
@@ -0,0 +1,581 @@
...
+        Log-Output -Message "DeviceInfo1: $($deviceInfo[1] | ConvertTo-Json -Depth 4)"
+    
+        return $deviceInfo[1]
+    }
+    
+    <#
+    .SYNOPSIS
+    Uses the current session with the PrivilegedEndpoint to enable remote management on device
+    #>
+    function Enable-RemoteManageOnDevice{
```
Comment (bganapa) [inline Registration/RemoteManagement.psm1:468]: Do we need a Disable- cmdlet?

**Review by bganapa — APPROVED** [review]

File: Registration/RemoteManagement.psm1 (hunk 3 above)
Comment (mtandon80) [reply Registration/RemoteManagement.psm1:468]: not right now. Disable is done directly from PEP. Documentation will call it out.

**Pull request merged**